src/components/App/App.js

```diff
@@ -53,16 +42,51 @@ function App() {
   };
 
   const handleLogout = () => {
+    setIsLoggedIn(false);
+    setCurrentUser(null);
+    setToken(null);
+    localStorage.removeItem("token");
+    setSavedArticles([]);
+  };
+
+  const fetchUserProfile = async (token) => {
     try {
-      localStorage.removeItem("isLoggedIn");
-      localStorage.removeItem("currentUser");
+      const userProfile = await getUserProfile(token);
+      setCurrentUser(userProfile); // Set user profile in state
     } catch (error) {
-      setError(`Logout Error: ${error.message}`);
-      console.error(`Logout Error: ${error.message}`);
+      setError(`Error fetching user profile: ${error.message}`);
+    }
+  };
+
+  const handleSignUp = async (signupData) => {
+    try {
+      await registerUser(signupData);
+      setSignUpOpen(false);
+      setConfirmationOpen(true);
+    } catch (error) {
+      setError(`Signup Error: ${error.message}`);
     }
-    setIsLoggedIn(false);
   };
 
+  const fetchSavedArticles = async (userToken) => {
+    try {
+      const articles = await getArticlesByUser(userToken);
+      setSavedArticles(articles);
+    } catch (error) {
+      setError(`Error fetching articles: ${error.message}`);
+    }
+  };
+
+  useEffect(() => {
+    const storedToken = localStorage.getItem("token");
+    if (storedToken) {
+      setToken(storedToken);
+      setIsLoggedIn(true);
+      fetchUserProfile(storedToken);
+      fetchSavedArticles(storedToken);
```

Comment:
` setIsLoggedIn(true);`

You can't set the user logged in if you find `storedToken` in `localStorage` because it can be **expired**. You need to first check the token on the server and in block `then` only set the user logged in

Reply:
You haven't corrected this remark